smolyay/surrogate.py

```diff
@@ -192,31 +192,52 @@ def gradient(self, x):
         """
         if self._coefficients is None:
             raise ValueError('Function needs training!')
-        x = numpy.array(x, copy=False, ndmin=1)
-        if x.shape != (self.dimension, ):
-            raise IndexError('Input must match dimension of domain.')
-        if not (numpy.all(x >= self._domain[:, 0])
-                and numpy.all(x <= self._domain[:, 1])):
+        input_shape = numpy.shape(x)
+        x = numpy.array(x, copy=False, ndmin=2)
+        if self.dimension == 1:
+            if len(input_shape) <= 1:
+                # the cast to 2d puts these in wrong order, so transpose
+                x = x.T
+            elif x.shape[-1] > 1:
+                x = x[..., numpy.newaxis]
+            elif input_shape == x.shape:
+                input_shape = input_shape[:-1]
```

Comment:
This is not related to getting `x` in the right shape, so it should be removed here. Its purpose is only for output, handle below.
```suggestion
```